inspirehep/modules/multieditor/actions.py

```diff
@@ -0,0 +1,282 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2017 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+from __future__ import absolute_import, print_function, division
+
+import re
+
+
+class Action(object):
+    def __init__(self, keys, value=None, match_type=None, value_to_check=None,
+                 conditions=None):
+        self.keys = keys
+        self.value = value
+        self.match_type = match_type
+        self.value_to_check = value_to_check
+        self.conditions = conditions
+        self.changed = False
+
+    def progress_keys(self, record, schema, position, checked):
+        key = self.keys[position]
+        condition_failed = False
+        new_schema = {}
+        if schema:  # for testing purposes
+            if schema['type'] == 'object':
+                new_schema = schema['properties'][key]
+            elif schema['type'] == 'array':
+                new_schema = schema['items']['properties'][key]
+
+        if self.conditions:
+            for condition in self.conditions:
+                if position < len(condition['key']) and key != condition['key'][position]\
+                        and (position == 0 or self.keys[position - 1] == condition['key'][position - 1]):
```

Comment:
In this `and`, the main key is equal to the condition key while on the comment you say 

```
# if the main key is different from the condition key for the first time
```